rover/src/main/kotlin/io/rover/rover/platform/DeviceIdentification.kt

```diff
@@ -0,0 +1,29 @@
+package io.rover.rover.platform
+
+import java.util.*
+
+interface DeviceIdentificationInterface {
+    val installationIdentifier: String
+}
+
+/**
+ * Responsible for maintaining an installation identifier.
+ */
+class DeviceIdentification(
+    localStorage: LocalStorage
+): DeviceIdentificationInterface {
+    private val storageContextIdentifier = "io.rover.rover.device-identification"
+    private val identifierKey = "identifier"
+    private val storage = localStorage.getKeyValueStorageFor(storageContextIdentifier)
+
+    override val installationIdentifier by lazy {
+        // Further reading: https://developer.android.com/training/articles/user-data-ids.html
+
+        // if persisted UUID not present then generate and persist a new one. Memoize it in memory.
+        storage.get(identifierKey) ?: (
+            UUID.randomUUID().toString().apply {
```

Comment:
How did you decide between using a randomly generated UUID and the InstanceID service provided by Google Play Services? Google seems to be suggesting the Instance ID unless it isn't "practical". 

> In cases where an Instance ID isn't practical, custom globally unique IDs (GUIDs) can also be used

Reply:
InstanceID is from Google's Play Services, which I understand we don't want to depend on in the Base SDK.

Reply:
Makes sense. I didn't know we could get away without depending on Play Services, but if that is an option it is ideal!